cfy_manager/main.py

```diff
@@ -543,45 +547,54 @@ def _print_finish_message(config_file=None):
 def _are_components_installed():
     return all(
         os.path.isfile(os.path.join(INITIAL_INSTALL_DIR, service_name))
-        for service_name in get_installed_services_names())
+        for service_name in config[SERVICES_TO_INSTALL])
 
 
 def _are_components_configured():
     return all(
         os.path.isfile(os.path.join(INITIAL_CONFIGURE_DIR, service_name))
-        for service_name in get_installed_services_names())
+        for service_name in config[SERVICES_TO_INSTALL])
 
 
 def is_supervisord_service():
     return service._get_service_type() == 'supervisord'
 
 
-def _create_initial_install_file():
+def _create_initial_install_file(installed_components_list):
     """
-    Create /etc/cloudify/.installed/<service_name> if the
-    service installation finished successfully for the first time
+    If the installation finished successfully for the first time:
+        1. create the file /etc/cloudify/.installed/<service_name>.
+        2. Update the file /etc/cloudify/.installed/components.yaml with the
+           dictionary {<main_service_name>: components_installed}.
+        3. Update the file /etc/cloudify/.installed/packages.yaml with the
+           dictionary {<main_service_name>: packages}.
     """
     if not _are_components_installed():
         mkdir(INITIAL_INSTALL_DIR)
-        for service_name in get_installed_services_names():
+        for service_name in config[SERVICES_TO_INSTALL]:
             touch(os.path.join(INITIAL_INSTALL_DIR, service_name))
+            if service_name in MAIN_SERVICES_NAMES:
```

Comment:
So if we don't keep them in a yaml file as well then that saves a few lines of code here.

Reply:
(marking as resolved as we discuss in separate thread)